cardano-testnet/test/Spec/Cli/Alonzo/LeadershipSchedule.hs

```diff
@@ -0,0 +1,527 @@
+{-# LANGUAGE BlockArguments #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DisambiguateRecordFields #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+
+{- HLINT ignore "Redundant return" -}
+{- HLINT ignore "Use let" -}
+
+module Spec.Cli.Alonzo.LeadershipSchedule
+  ( hprop_leadershipSchedule
+  ) where
+
+import           Cardano.Api (AlonzoEra, SerialiseAddress (serialiseAddress), UTxO (UTxO))
+import qualified Cardano.Api as Api
+import           Cardano.Api.Shelley (PoolId)
+import           Cardano.CLI.Shelley.Output (QueryTipLocalStateOutput (mEpoch))
+import           Cardano.CLI.Shelley.Run.Query (DelegationsAndRewards, mergeDelegsAndRewards)
+import           Control.Monad (void)
+import           Data.Aeson (FromJSON)
+import qualified Data.Aeson.Types as J
+import           Data.List ((\\))
+import qualified Data.List as L
+import qualified Data.Map.Strict as Map
+import           Data.Monoid (Last (..))
+import           Data.Set (Set)
+import qualified Data.Set as Set
+import           Data.Text (Text)
+import qualified Data.Text as T
+import qualified Data.Time.Clock as DTC
+import           GHC.Generics (Generic)
+import           GHC.Stack (callStack)
+import           Hedgehog (Property, (===))
+import qualified Hedgehog as H
+import qualified Hedgehog.Extras.Stock.IO.Network.Sprocket as IO
+import qualified Hedgehog.Extras.Test.Base as H
+import qualified Hedgehog.Extras.Test.Concurrent as H
+import qualified Hedgehog.Extras.Test.File as H
+import qualified Hedgehog.Extras.Test.Process as H
+import           Prelude
+import qualified System.Directory as IO
+import           System.Environment (getEnvironment)
+import           System.FilePath ((</>))
+import qualified System.Info as SYS
+import           System.Info (os)
+import qualified Test.Assert as H
+import qualified Test.Base as H
+import qualified Test.Process as H
+import qualified Test.Runtime as TR
+import qualified Testnet.Cardano as TC
+import           Testnet.Cardano (TestnetOptions (..), TestnetRuntime (..))
+import qualified Testnet.Conf as H
+import           Testnet.Utils (waitUntilEpoch)
+
+isLinux :: Bool
```

Comment:
Let's move this to the utils module in `cardano-testnet` because this has already been defined

Reply:
Done